File: apps/frontend-react-vite/src/types/link.ts
export interface Link {
  id: string
  name: string
  url: string
  group_id?: string
  flow_id?: string | null
  order_index: number
  created_at?: string
  updated_at?: string
}

export interface LinkCreate {
  name: string
  url: string
  group_id?: string
  flow_id?: string | null
  order_index?: number
}

export interface LinkUpdate {
  name?: string
  url?: string
  group_id?: string
  flow_id?: string | null
  order_index?: number
}

export interface LinkGroup {
  id: string
  name: string
  description?: string
  order_index: number
  created_at?: string
  updated_at?: string
}

export interface LinkGroupCreate {
  name: string
  description?: string
  order_index?: number
}

export interface LinkGroupUpdate {
  name?: string
  description?: string
  order_index?: number
}

export interface LinkFlow {
  id: string
  group_id: string
  name: string
  order_index: number
  created_at?: string
  updated_at?: string
}

export interface LinkFlowCreate {
  group_id: string
  name: string
  order_index?: number
}

export interface LinkFlowUpdate {
  group_id?: string
  name?: string
  order_index?: number
}

export interface LinkFlowWithLinks {
  flow: LinkFlow
  links: Link[]
}

export interface GroupedLinks {
  group: LinkGroup
  flows: LinkFlowWithLinks[]
  links: Link[]
}

File: apps/backend-go-fiber/internal/model/link.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Link struct {
	ID         uuid.UUID  `gorm:"type:text;primaryKey" json:"id"`
	Name       string     `gorm:"not null" json:"name"`
	URL        string     `gorm:"not null" json:"url"`
	GroupID    *uuid.UUID `gorm:"type:text" json:"group_id,omitempty"`
	FlowID     *uuid.UUID `gorm:"type:text" json:"flow_id,omitempty"`
	OrderIndex int        `gorm:"default:0" json:"order_index"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
}

func (l *Link) BeforeCreate(tx *gorm.DB) error {
	if l.ID == uuid.Nil {
		l.ID = uuid.New()
	}
	return nil
}

type LinkCreate struct {
	Name       string  `json:"name" validate:"required"`
	URL        string  `json:"url" validate:"required"`
	GroupID    *string `json:"group_id,omitempty"`
	FlowID     *string `json:"flow_id,omitempty"`
	OrderIndex int     `json:"order_index"`
}

type LinkUpdate struct {
	Name       *string `json:"name,omitempty"`
	URL        *string `json:"url,omitempty"`
	GroupID    *string `json:"group_id,omitempty"`
	FlowID     *string `json:"flow_id,omitempty"`
	OrderIndex *int    `json:"order_index,omitempty"`
}
